.gitignore

```diff
@@ -398,3 +398,5 @@ cmake-build*/
 # vcpkg
 !make.exe
 vcpkg_installed/
+
+compile_commands.json
```

Comment:
Is this block needed? I can't see any file committed with this name. Thanks :)

Reply:
It's in gitignore to avoid being added into the git index. I have it locally because I used neovim

Reply:
Ah ok I see, let me add one more patch on top then, we're aligned with https://github.com/github/gitignore as this file is mainly a combination of different ignore files.

Reply:
Done, your line is now part of the default CMake ignore file :) I used this chance to update the others as well as adding the Jetbrain one to ensure wider IDEs support. Let me know if you're fine with this change. Thanks!

Reply:
LGTM. Thanks!